lib/Doctrine/DBAL/Platforms/DB2iSeriesPlatform.php

```diff
@@ -0,0 +1,312 @@
+    public function getListTablesSQL($database = null)
+    {
+        return "
+            SELECT
+              DISTINCT NAME
+            FROM
+                SYSIBM.tables t
+            WHERE
+              table_type='BASE TABLE'
+              ". (!is_null($database) ? "AND t.TABLE_SCHEMA = UPPER('$database')" : '') ."
+            ORDER BY NAME
+        ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getListViewsSQL($database = null)
+    {
+        return "
+            SELECT
+              DISTINCT NAME,
+              TEXT
+            FROM QSYS2.sysviews v
+            WHERE 1=1
+            ". (!is_null($database) ? "AND v.TABLE_SCHEMA = UPPER('$database')" : '') ."
+            ORDER BY NAME
+        ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getListTableIndexesSQL($table, $database = null)
+    {
+        return  "
+            SELECT
+                  scc.CONSTRAINT_NAME as key_name,
+                  scc.COLUMN_NAME as column_name,
+                  CASE
+                      WHEN sc.CONSTRAINT_TYPE = 'PRIMARY KEY' THEN 1
+                      ELSE 0
+                  END AS primary,
+                  CASE
+                      WHEN sc.CONSTRAINT_TYPE = 'UNIQUE' THEN 0
+                      ELSE 1
+                  END AS non_unique
+              FROM
+              QSYS2.syscstcol scc
+              LEFT JOIN QSYS2.syscst sc ON
+                  scc.TABLE_SCHEMA = sc.TABLE_SCHEMA AND scc.TABLE_NAME = sc.TABLE_NAME AND scc.CONSTRAINT_NAME = sc.CONSTRAINT_NAME
+            WHERE scc.TABLE_NAME = UPPER('" . $table . "')
+            ". (!is_null($database) ? "AND scc.TABLE_SCHEMA = UPPER('$database')" : '') ."
+        ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getListTableForeignKeysSQL($table)
+    {
+        return "
+            SELECT DISTINCT
+                fk.COLUMN_NAME AS local_column,
+                pk.TABLE_NAME AS foreign_table,
+                pk.COLUMN_NAME AS foreign_column,
+                fk.CONSTRAINT_NAME AS index_name,
+                rc.UPDATE_RULE AS on_update,
+                rc.DELETE_RULE AS on_delete
+            FROM QSYS2.REFERENTIAL_CONSTRAINTS rc
+            LEFT JOIN QSYS2.SYSCSTCOL fk ON
+                rc.CONSTRAINT_SCHEMA = fk.CONSTRAINT_SCHEMA AND
+                rc.CONSTRAINT_NAME = fk.CONSTRAINT_NAME
+            LEFT JOIN QSYS2.SYSCSTCOL pk ON
+                rc.UNIQUE_CONSTRAINT_SCHEMA = pk.CONSTRAINT_SCHEMA AND
+                rc.UNIQUE_CONSTRAINT_NAME = pk.CONSTRAINT_NAME
+            WHERE fk.TABLE_NAME = UPPER('" . $table . "')
+        ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getCreateDatabaseSQL($database)
+    {
+        return "CREATE COLLECTION ".$database;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doModifyLimitQuery($query, $limit, $offset = null)
+    {
+        if ($limit === null && $offset === null) {
```

Comment:
I took most of the logic from the SQLServerPlatform-> doModifyLimitQuery.
I'm already trying to simplify the method but there's lot of stuff going on there, so for now is the best i can get. Unfortunatly not my code, but it gets the job done.